src/matrix/kaldi-vector.h

```diff
@@ -119,6 +120,15 @@ class VectorBase {
   template<typename OtherReal>
   void CopyFromVec(const CuVectorBase<OtherReal> &v);
 
+  /// Applies floor to all elements. Returns number of elements
+  /// floored in floored_count if it is non-null.
```

Comment:
The only part of this PR that I am uncertain about is the floor/ceiling stuff.  I guess I'm concerned about the extra memory bandwidth from writing the same thing back to memory in case this and v point to the same data (v.Data() == this->data_).  Perhaps there could be an if/else inside the floor/ceiling functions to take care of that special case?  Also the documentation needs to be changed, it's not accurate now that the function changed.  Also (this was a pre-existing issue), the word "returns" is not appropriate there, it should be "outputs"